File: Leetcode/88. Merge Sorted Array.cpp
class Solution {
public:
    void merge(int A[], int m, int B[], int n) {
        int i,j,k;
        if(m>0)
        for(i=m-1;i>=0;i--)
        A[i+n]=A[i];
        if(m==0)
        {
            for(i=0;i<n;i++)
            A[i]=B[i];
            return;
        }
        if(n==0)
        {
            return;
        }
        k=0;
        i=0;j=n;
        while(i<n && j<m+n)
        {
            if(B[i]<A[j])
            {
                A[k]=B[i];
                k++;
                i++;
            }
            else
            {
                A[k]=A[j];
                k++;
                j++;
            }
        }
        while(i<n)
        {
            A[k]=B[i];
            k++;
            i++;
            
        }
        while(j<m+n)
        {
            A[k]=A[j];
            k++;
            j++;
        }
    }
};

File: Leetcode/88. Merge Sorted Array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        
        if(0==n){
            return;
        }
        int i,j,k;
        i = m-1; j=n-1;k=m+n-1;
        while(i>=0 && j>=0){
            if(nums1[i] > nums2[j]){
                nums1[k] = nums1[i];
                i--;
            }else{
                nums1[k] = nums2[j];
                j--;
            }
            k--;
        }
        if(i>=0){
            while(i>=0){
                nums1[k] = nums1[i];
                i--;
                k--;
            }
        }else if(j>=0){
            while(j>=0){
                nums1[k] = nums2[j];
                j--;
                k--;
            }
        }
    }
}
